### PR TITLE
INFINITY-2967 Enable cassandra agent shutdown test as the last test to run

### PR DESCRIPTION
**Pull request by nickbp**
This test is important enough to keep in PRs if possible.
It's a destructive test, but if it's the last test to be run then that doesn't matter.

### CODE REVIEW

**Comment by nickbp** [comment]
Imported @gabrielhartmann's changes to host detection from #2090

**Comment by nickbp** [comment]
It does look like there's an underlying issue, as hosts do not align between the output of `dcos hosts` vs the TaskInfo labels:
```
[2018-01-12 23:37:05,587|sdk_cmd|INFO]: STDOUT:
NAME           HOST         USER   STATE  ID                                                   MESOS ID                                     REGION          ZONE       
cassandra      10.0.1.201   root     R    cassandra.019bdc3b-f7f1-11e7-a89c-2a718af34440       5c2ccdd4-cf9c-4b9f-b692-e34f73db72a2-S3  aws/us-west-2  aws/us-west-2a  
node-0-server  10.0.1.201  nobody    R    node-0-server__f60a5aa7-7fa5-4007-b7c3-988d5d8336dc  5c2ccdd4-cf9c-4b9f-b692-e34f73db72a2-S3  aws/us-west-2  aws/us-west-2a  
node-1-server  10.0.3.246  nobody    R    node-1-server__eebeb1fc-1f91-4d61-9818-4dea935710b6  5c2ccdd4-cf9c-4b9f-b692-e34f73db72a2-S1  aws/us-west-2  aws/us-west-2a  
node-2-server  10.0.1.71   nobody    R    node-2-server__ef5ddd9f-cd2c-472f-ae77-3534e93be426  5c2ccdd4-cf9c-4b9f-b692-e34f73db72a2-S2  aws/us-west-2  aws/us-west-2a
[2018-01-12 23:37:05,653|tests.test_zzzrecovery|INFO]: Scheduler IP: 10.0.1.201
[2018-01-12 23:37:08,107|tests.test_zzzrecovery|INFO]: Pods:
{'node-0': {'agent': '5c2ccdd4-cf9c-4b9f-b692-e34f73db72a2-S2',
            'host': '10.0.1.71',
            'name': 'node-0'},
 'node-1': {'agent': '5c2ccdd4-cf9c-4b9f-b692-e34f73db72a2-S1',
            'host': '10.0.3.246',
            'name': 'node-1'},
 'node-2': {'agent': '5c2ccdd4-cf9c-4b9f-b692-e34f73db72a2-S2',
            'host': '10.0.1.71',
            'name': 'node-2'}}
[2018-01-12 23:37:08,108|tests.test_zzzrecovery|INFO]: Found pod avoiding scheduler at 10.0.1.201: {'agent': '5c2ccdd4-cf9c-4b9f-b692-e34f73db72a2-S2', 'host': '10.0.1.71', 'name': 'node-2'}
```
AFAIK we just copy the host/agentid information from Mesos. Going to add validation which compares the two data sources, and fails the test if they don't match.

File: frameworks/cassandra/tests/test_zzzrecovery.py [hunk 1]
```diff
@@ -58,41 +60,65 @@ def test_node_replace_replaces_node():
...
+    replace_pod = None
+    for key, value in pods.items():
+        if value['host'] != scheduler_ip:
+            replace_pod = value
+            log.info('Found pod avoiding scheduler at {}: {}'.format(scheduler_ip, value))
+            break
+    return replace_pod
 
 
 def get_pod_agent(pod_name):
```
Comment (gabrielhartmann) [inline frameworks/cassandra/tests/test_zzzrecovery.py]: `get_pod_agent` and `get_pod_host` cannot assume the first task in the list of tasks for a pod is representative of all tasks for the pod.  Instead they should look for the `server` task's information.

**Comment by gabrielhartmann** [comment]
*Copying paraphrased Slack conversation to the PR for posterity*


It’s a test bug.  Relying on every task in a pod having the same host/agent info is wrong.

The `get_pod_agent` and `get_pod_host` commands cannot assume that the first element in the task list for a pod is representative of the whole running pod.  They should instead use only the `server` task to determine the pod's location.

Let's look at `node-0` in particular. It has as it’s first task `node-0-init_system_keyspaces`.  This has a goal state of ONCE.

1. Assume it gets launched and completes successfully on agent `A`.
1. We then replace `node-0`, and `node-0` lands on agent `B`.
1. Then we look up `node-0`’s host/agent info based on `node-0-init_system_keyspaces`.

There’s the mistake.  That task was launched on Agent `A` and so returns Agent `A`’s info although the `server` task and therefore the pod is running on Agent `B`.

**Pull request merged**